### PR TITLE
Add support for `QueryStatus` field

### PR DESCRIPTION
**Pull request by tiratatp**
### Why is this change required?
- We recently released a `QueryStatus` field that contains a current
query status.

### Does this commit cause any expected behaviors to change? If so, what?
- None

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: sample_apps/dotnet/README.md [hunk 1]
```diff
@@ -10,19 +10,19 @@ Sample .Net application to use Timestream
 
 1. Install required NuGet. Ensure AWSSDK.Core version is 3.3.107 or newer.
    ```shell
-   dotnet add package AWSSDK.Core
-   dotnet add package AWSSDK.TimestreamWrite
-   dotnet add package AWSSDK.TimestreamQuery
+   dotnet add package AWSSDK.Core --version 3.5.1.41
+   dotnet add package AWSSDK.TimestreamWrite --version 3.5.0.20
```
Comment (rajeiyer) [inline sample_apps/dotnet/README.md:14]: Why not use the same versions for both the write and query SDKs

**Review by rajeiyer — COMMENTED** [review]
Nit: minor comment around SDK versions. Otherwise LGTM.

File: sample_apps/dotnet/README.md (hunk 1 above)
Comment (tiratatp) [reply sample_apps/dotnet/README.md:14]: The latest version for each of these components are different:
- https://www.nuget.org/packages/AWSSDK.Core/
- https://www.nuget.org/packages/AWSSDK.TimestreamWrite/
- https://www.nuget.org/packages/AWSSDK.TimestreamQuery/

**Review by rajeiyer — APPROVED** [review]

File: sample_apps/python/QueryExample.py [hunk 2]
```diff
@@ -273,6 +273,17 @@ def run_query(self, query_string):
...
+        query_status = query_result["QueryStatus"]
+
+        progress_percentage = query_status["ProgressPercentage"]
+        print(f"Query progress so far: {progress_percentage}%")
+
+        bytes_scanned = float(query_status["CumulativeBytesScanned"]) / ONE_GB_IN_BYTES
+        print(f"Bytes Scanned so far: {bytes_scanned} GB")
+
+        bytes_metered = float(query_status["CumulativeBytesMetered"]) / ONE_GB_IN_BYTES
+        print(f"Bytes Metered so far: {bytes_metered} GB")
```
Comment (ozanozen) [inline sample_apps/python/QueryExample.py:285]: again bytes vs data

File: sample_apps/java/src/main/java/com/amazonaws/services/timestream/QueryExample.java [hunk 3]
```diff
@@ -283,6 +286,16 @@ private void runQuery(String queryString) {
     }
 
     private void parseQueryResult(QueryResult response) {
+        final QueryStatus queryStatus = response.getQueryStatus();
+
+        System.out.println("Query progress so far: " + queryStatus.getProgressPercentage() + "%");
+
+        double bytesScannedSoFar = (double) queryStatus.getCumulativeBytesScanned() / ONE_GB_IN_BYTES;
+        System.out.println("Bytes scanned so far: " + bytesScannedSoFar + " GB");
```
[the remark this answers is not in the document]
Comment (tiratatp) [reply sample_apps/java/src/main/java/com/amazonaws/services/timestream/QueryExample.java:294]: The fields that we have are `CumulativeBytesScanned` and `CumulativeBytesMetered`.

Maybe I could list out number in bytes, and again in gigabytes?

**Pull request merged**